File: Structural/Adapter/CSharp/MP3Player.cs
// MP3Player.cs
using System;

public class MP3Player : IMediaPlayer {
  private MediaAdapter? mediaAdapter;

  public void Play(string audioType, string fileName) {
    if (audioType.Equals("MP3", StringComparison.OrdinalIgnoreCase)) {
      Console.WriteLine("Playing MP3 file: " + fileName);
    } else if (audioType.Equals("MP4", StringComparison.OrdinalIgnoreCase) || 
              audioType.Equals("VLC", StringComparison.OrdinalIgnoreCase)) {
      mediaAdapter = new MediaAdapter(audioType);
      mediaAdapter.Play(audioType, fileName);
    } else {
      Console.WriteLine("Invalid media type: " + audioType);
    }
  }
}

File: Structural/Adapter/Cpp/MP3Player.cpp
// MP3Player.cpp
#include "MP3Player.h"

void MP3Player::play(const std::string& audioType, const std::string& fileName) {
  if (audioType == "MP3") {
    std::cout << "Playing MP3 file: " << fileName << std::endl;
  } else if (audioType == "MP4" || audioType == "VLC") {
    mediaAdapter = std::make_unique<MediaAdapter>(audioType);
    mediaAdapter->play(audioType, fileName);
  } else {
    std::cout << "Invalid media type: " << audioType << std::endl;
  }
}
